contracts/truefi2/LoanFactory2.sol

```diff
@@ -40,6 +41,7 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
     ITrueRateAdjuster public rateAdjuster;
     ITrueFiCreditOracle public creditOracle;
     IBorrowingMutex public borrowingMutex;
+    address public loanTokenImplementation;
```

Comment:
Can we allow the compiler to typecheck this for us? We would declare it in the contract as `ILoanToken2 public loanTokenImplementation` but then need to call `Clones.clone(address(loanTokenImplementation))` when passing it to a proxy later.

This would give us more of a safeguard in our tests and deploy scripts to make sure we're not passing in (for instance) a deprecated ILoanToken or IDeficiencyToken by accident. And if we do a LoanToken2 refactor or upgrade, we won't forget to check whether that breaks the build in this contract.

Of course, declaring the correct type here doesn't prevent us from passing in the wrong address on mainnet :)

Reply:
In case I'm not awake to re-approve, this PR LGTM after this fix.